core/src/main/java/com/linecorp/armeria/internal/Http1ObjectEncoder.java

```diff
@@ -116,31 +120,43 @@ protected Channel channel() {
     }
 
     @Override
-    protected ChannelFuture doWriteHeaders(int id, int streamId, HttpHeaders headers, boolean endStream) {
+    protected ChannelFuture doWriteHeaders(int id, int streamId, HttpHeaders headers, boolean endStream,
+                                           HttpHeaders additionalHeaders, HttpHeaders additionalTrailers,
+                                           boolean enableServerHeader, boolean enableDateHeader,
```

Comment:
`enableServerHeader`, `enableDateHeader` is immutable after building `Server`. 
I think it would be nice to get these flags from class fields that were set when creating `Http*ObjectEncoder`.